File: fixtures/java/jca/hmac-sha256/Main.java
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;

public class Main {
    public static void main(String[] args) throws Exception {
        byte[] key = "secret_key".getBytes();
        byte[] message = "Hello, World!".getBytes();
        
        // Create HMAC
        Mac mac = Mac.getInstance("HmacSHA256");
        SecretKeySpec keySpec = new SecretKeySpec(key, "HmacSHA256");
        mac.init(keySpec);
        byte[] hmac = mac.doFinal(message);
        
        // Verify HMAC (compare with expected)
        Mac verifier = Mac.getInstance("HmacSHA256");
        verifier.init(keySpec);
        byte[] expectedHmac = verifier.doFinal(message);
    }
}

File: fixtures/kotlin/jca/hmac-sha256/Main.kt
import javax.crypto.Mac
import javax.crypto.spec.SecretKeySpec

fun main() {
    val key = "secret_key".toByteArray()
    val message = "Hello, World!".toByteArray()
    
    // Create HMAC
    val mac = Mac.getInstance("HmacSHA256")
    val keySpec = SecretKeySpec(key, "HmacSHA256")
    mac.init(keySpec)
    val hmac = mac.doFinal(message)
    
    // Verify HMAC
    val verifier = Mac.getInstance("HmacSHA256")
    verifier.init(keySpec)
    val expectedHmac = verifier.doFinal(message)
}
